confmap/confmap.go

```diff
@@ -209,3 +209,11 @@ func mapKeyStringToMapKeyTextUnmarshalerHookFunc() mapstructure.DecodeHookFuncTy
 		return data, nil
 	}
 }
+
+// Unmarshallable defines an optional interface for custom configuration unmarshalling.
+// A configuration struct can implement this interface to override the default unmarshalling.
+type Unmarshallable interface {
```

Comment:
s/Unmarshallable/Unmarshaler I think this is the standard that almost all the golang is using. I know the name is my fault, but good opportunity to fix it.